demos/mantle_convection/free_surface/free_surface.py

```diff
@@ -124,11 +124,11 @@
 # Typically, when the exterior density is negligbile compared with the interior density (i.e. air vs rock!) and with $\alpha$ = 3x10$^{-5}$ K$^{-1}$ and $\Delta T$ = 3000 K, $B_{fs}$ is about 10. For dimensional simulations you can specify the density contrast across the free surface with *delta_rho_fs* through the free surface dictionary, as well as providing $\rho_0$ and $g$ through the approximation object in the usual way.
 
 # +
-Bfs = 10.  # Free surface buoyancy number
+Bfs = 10.0  # Free surface buoyancy number
 
 stokes_bcs = {
     boundary.bottom: {"uy": 0},
-    boundary.top: {"free_surface": {"RaFS": Ra*Bfs}},
+    boundary.top: {"free_surface": {"eta_index": 2, "RaFS": Ra * Bfs}},
```

Comment:
I haven't had time to look at this, but I'm not sure I like the ergonomics of this `eta_index` mechanism... Let me have a proper look and think first though.

Reply:
Thanks! I disagree that it's fragile to rely on insertion order: it's an explicit feature of Python 3.7+ and we're well past that (we should probably nuke the comments about that...) It is true that you need to be a bit more careful setting things up with multiple free surface BCs, true.

With that in mind, my preference would be not to introduce `eta_index` here. I definitely agree that *something* is needed, particularly when internal variables are introduced, but I don't think it's time just yet.

Reply:
No wait, you are missing the point. I am not talking about the dictionary insertion order (I agree with your conclusion on that). I am talking about the fact that the insertion order in the dictionary has to match the order of the functions in the mixed space.

Reply:
Perhaps another way to deal with that is to add a boundary attribute to each free-surface function in the same way that we add the cartesian attribute to the mesh?

Reply:
They're both good ideas, and possibilities for sure. My preference is to make sure we think about this properly, so we get a clean and ergonomic interface. It's not in scope for this PR in particular.

Reply:
Given that there is more than one possibility now, I agree that the way forward is to leave this change out and open an issue mentioning the problem and potential solutions.